### PR TITLE
Count incomplete upgrades in VersionStatus

### PR DESCRIPTION
**Pull request by mpolden**

### CODE REVIEW

**Comment by mpolden** [comment]
Should probably only count consecutive failed runs. Will fix.

**Comment by mpolden** [comment]
PTAL

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java [hunk 1]
```diff
@@ -202,13 +201,24 @@ private static NodeVersions findSystemApplicationVersions(Controller controller)
...
+                for (var run : runsOfFailingJobs) {
+                    if (run.status() == RunStatus.success) {
+                        consecutiveFailingRuns.clear();
+                    } else {
+                        consecutiveFailingRuns.add(run);
+                    }
+                }
+                consecutiveFailingRuns.stream().map(run -> run.versions().targetPlatform())
+                                      .forEach(version -> versionMap.put(version, versionMap.getOrDefault(version, DeploymentStatistics.empty(version))
+                                                                                            .withFailing(id)));
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java]: How about
```java
                jobs.failing()
                    .not().failingApplicationChange()
                    .not().withStatus(RunStatus.outOfCapacity)
                    .mapToList(JobStatus::runs)
                    .forEach(runs -> runs.descendingMap().values().stream()
                                         .dropWhile(run -> ! run.hasEnded())
                                         .takeWhile(run -> run.hasFailed())
                                         .map(run -> run.versions().targetPlatform())
                                         .forEach(version -> versionMap.put(version,
                                                                            versionMap.getOrDefault(version, DeploymentStatistics.empty(version))
                                                                                      .withFailing(id))));
```

the jobs contains runs for all job types, each of which may be relevant — not just the consecutive failures for the last job type. 

Also, jobs which are running (not `success`) aren't failures.

Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java]: Thanks, I got lost in the *List jungle when I tried a different approach. 😅

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/versions/VersionStatus.java]: Haha, I know. I get lost myself >_<

**Review by jonmv — APPROVED** [review]

**Comment by mpolden** [comment]
Java build passed.

**Pull request merged**